Panic when fuse fails to start

When the fuse binary is missing, the user doesn't have read access, user_allow_other isn't set, etc., the Proxy will panic instead of reporting the error and exiting gracefully.

Stacktrace:

```
enocom@instance-1:~$ ./cloud-sql-proxy --fuse /cloudsql
2023/10/31 19:42:46 Authorizing with Application Default Credentials
2023/10/31 19:42:46 The proxy has started successfully and is ready for new connections!
/usr/bin/fusermount: option allow_other only allowed if 'user_allow_other' is set in /etc/fuse.conf
2023/10/31 19:42:46 The proxy has encountered a terminal error: FUSE mount failed: "/cloudsql": fusermount exited with code 256
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xbc3052]

goroutine 1 [running]:
github.com/hanwen/go-fuse/v2/fuse.(*Server).Unmount(0x658bb9?)
        /workspace/GOPATH/pkg/mod/github.com/hanwen/go-fuse/v2@v2.4.0/fuse/server.go:128 +0x12
github.com/GoogleCloudPlatform/cloud-sql-proxy/v2/internal/proxy.(*Client).unmountFUSE(0x0?)
        /workspace/internal/proxy/proxy_other.go:183 +0x85
github.com/GoogleCloudPlatform/cloud-sql-proxy/v2/internal/proxy.(*Client).Close(0xc000564280)
        /workspace/internal/proxy/proxy.go:647 +0x5c
github.com/GoogleCloudPlatform/cloud-sql-proxy/v2/cmd.runSignalWrapper.func7()
        /workspace/cmd/root.go:821 +0x2d
github.com/GoogleCloudPlatform/cloud-sql-proxy/v2/cmd.runSignalWrapper(0xc000632780)
        /workspace/cmd/root.go:911 +0x160c
github.com/GoogleCloudPlatform/cloud-sql-proxy/v2/cmd.NewCommand.func3(0xc00057e300, {0xc000500fa0, 0x7, 0xf6388a})
        /workspace/cmd/root.go:361 +0x17
github.com/spf13/cobra.(*Command).execute(0xc00057e300, {0xc00003e250, 0x2, 0x2})
        /workspace/GOPATH/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:940 +0x87c
github.com/spf13/cobra.(*Command).ExecuteC(0xc00057e300)
        /workspace/GOPATH/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(...)
        /workspace/GOPATH/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
github.com/GoogleCloudPlatform/cloud-sql-proxy/v2/cmd.Execute()
        /workspace/cmd/root.go:76 +0x25
main.main()
        /workspace/main.go:25 +0xf
```

--- a/internal/proxy/proxy_other.go
+++ b/internal/proxy/proxy_other.go
@@ -184,6 +184,9 @@ func (c *Client) fuseMounts() []*socketMount {
 func (c *Client) unmountFUSE() error {
 	c.fuseServerMu.Lock()
 	defer c.fuseServerMu.Unlock()
+	if c.fuseServer == nil {
+		return nil
+	}
 	return c.fuseServer.Unmount()
 }
 

--- a/internal/proxy/proxy_other_test.go
+++ b/internal/proxy/proxy_other_test.go
@@ -18,8 +18,11 @@
 package proxy_test
 
 import (
+	"context"
 	"os"
 	"testing"
+
+	"github.com/GoogleCloudPlatform/cloud-sql-proxy/v2/internal/proxy"
 )
 
 var (
@@ -39,4 +42,21 @@ func verifySocketPermissions(t *testing.T, addr string) {
 	if fm := fi.Mode(); fm != 0777|os.ModeSocket {
 		t.Fatalf("file mode: want = %v, got = %v", 0777|os.ModeSocket, fm)
 	}
+}
+
+func TestFuseClosesGracefully(t *testing.T) {
+	c, err := proxy.NewClient(
+		context.Background(), nil, testLogger,
+		&proxy.Config{
+			FUSEDir:     t.TempDir(),
+			FUSETempDir: t.TempDir(),
+			Token:       "mytoken",
+		})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := c.Close(); err != nil {
+		t.Fatal(err)
+	}
+
 }

